### PR TITLE
[SPARK-9030][STREAMING] Add Kinesis.createStream unit tests that actual sends data

### PR DESCRIPTION
**Pull request by tdas**
Current Kinesis unit tests do not test createStream by sending data. This PR is to add such unit test. Note that this unit will not run by default. It will only run when the relevant environment variables are set.

### CODE REVIEW

**Comment by tdas** [comment]
Jenkins, test this please.

**Comment by tdas** [comment]
Jenkins, test this please.

File: extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        .withData(ByteBuffer.wrap(str.getBytes()))
+        .withPartitionKey(str)
+
+      val putRecordResult = kinesisClient.putRecord(putRecordRequest)
+      val shardId = putRecordResult.getShardId
+      val seqNumber = putRecordResult.getSequenceNumber()
+      val sentSeqNumbers = shardIdToSeqNumbers.getOrElseUpdate(shardId,
+        new ArrayBuffer[(Int, String)]())
+      sentSeqNumbers += ((num, seqNumber))
+
```
Comment (zsxwing) [inline extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala]: extra line

File: extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisStreamSuite.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  after {
+    if (ssc != null) {
+      ssc.stop(stopSparkContext = false)
+      ssc = null
+    }
+  }
+
+  test("API") {
```
Comment (zsxwing) [inline extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisStreamSuite.scala]: Isn't this one same as `KinesisReceiverSuite.KinesisUtils API`?

File: extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    RegionUtils.getRegionByEndpoint(endpointUrl).getName()
+  } else {
+    RegionUtils.getRegion(_regionName).getName()
+  }
+  val streamShardCount = 2
+
+  private val createStreamTimeoutSeconds = 300
+  private val describeStreamPollTimeSeconds = 1
+
+  @volatile
```
Comment (zsxwing) [inline extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala]: Is `@volatile` necessary? Looks `streamCreated` is only used in one thread.

**Comment by zsxwing** [comment]
Thanks @tdas I'm not familiar with Kinesis client APIs. But generally looks good and Jenkins is green. I will add Python tests for the Kinesis Python API after this PR is merged.

File: extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisStreamSuite.scala (hunk 2 above)
Comment (tdas) [inline extras/kinesis-asl/src/test/scala/org/apache/spark/streaming/kinesis/KinesisStreamSuite.scala]: Yes, I forgot to remove that unit test, good point.

File: extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala (hunk 3 above)
Comment (tdas) [inline extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisTestUtils.scala]: Just making it threadsafe in general

**Comment by tdas** [comment]
Jenkins, test this please.

**Comment by zsxwing** [comment]
LGTM

**Comment by tdas** [comment]
Thanks @zsxwing, merging this.

**Pull request closed without merging**